superset-frontend/src/SqlLab/components/TabbedSqlEditors.jsx

```diff
@@ -76,6 +85,7 @@ class TabbedSqlEditors extends React.PureComponent {
       queriesArray: [],
       dataPreviewQueries: [],
       hideLeftBar: false,
+      databaseIds: getDatabasesIds(),
```

Comment:
Mmmh, we're already pulling the database list to populate the dropdown, can we just use/share that? Wondering why it used to work and doesn't anymore (?)

It seems to me like this belongs somewhere in redux land.

Reply:
Currently the database selector is using `<SupersetAsyncSelect/>` which doesn't allow the TabbedSqlEditors to have a reference for the databases available. So i'm basically following the same pattern but in our component `makeApi`.

This is blocking for this weeks release so I wanted to fix the issue, and then I can look into moving this code into redux.

https://github.com/apache/incubator-superset/blob/master/superset-frontend/src/components/DatabaseSelector.tsx#L222

Reply:
I'm ok with this as a quick fix to unblock if we revisit quickly. But I don't think fetching in components is great, especially in constructors as it will hold things up.

Pointing out two relevant redux actions that could be useful for the following PR:

https://github.com/apache/incubator-superset/blob/master/superset-frontend/src/SqlLab/actions/sqlLab.js#L436

https://github.com/apache/incubator-superset/blob/master/superset-frontend/src/SqlLab/actions/sqlLab.js#L744